sklearn/feature_extraction/text.py

```diff
@@ -714,7 +714,7 @@ def fit(self, X, y=None):
                 "string object received.")
 
         self._validate_params()
-
+        self.build_analyzer()
```

Comment:
Add a comment to say that this is used for further validation

Reply:
done